tests/bsim/bluetooth/host/misc/sample_test/src/peer.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/att.h>
+#include <zephyr/bluetooth/gatt.h>
+
+#include "testlib/adv.h"
+#include "testlib/att_read.h"
+#include "testlib/att_write.h"
+#include "testlib/conn.h"
+
+#include "bs_helpers/flags.h"
+#include "bs_helpers/sync.h"
+#include "bs_helpers/testcase.h"
+#include "bs_helpers/timeout.h"
+
+/* local includes */
+#include "data.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(peer, CONFIG_TEST_APP_LOG_LEVEL);
+
+static DEFINE_FLAG(is_subscribed);
+static DEFINE_FLAG(got_notification_1);
+static DEFINE_FLAG(got_notification_2);
+
+int find_characteristic(struct bt_conn *conn,
+			const struct bt_uuid *svc,
+			const struct bt_uuid *chrc,
+			uint16_t *chrc_value_handle)
+{
+	uint16_t svc_handle;
+	uint16_t svc_end_handle;
+	uint16_t chrc_end_handle;
+	int err;
+
+	LOG_DBG("");
+
+	err = bt_testlib_gatt_discover_primary(&svc_handle, &svc_end_handle, conn, svc, 1, 0xffff);
+	if (err) {
```

Comment:
Ditto with using boolean values for boolean expressions

Reply:
cmon, we have to ditch the whole of zephyr then. This is pretty much the idiomatic error handing in zephyr.

Reply:
Indeed. The coding guidelines do state this though, but it also states that we are not yet at a level where this should be enforced, so it's not a blocking comment, but the plan is actually to modify all `if` statements in Zephyr to comply with that rule (at some point). 

My personal opinion is that we might as well follow the to-be-followed rules now for new code

Reply:
> Do we have a task-force for MISRA-ing the tree or is the plan to only enforce the rule on new PRs?

I think it's just been a low-effort thing from since like 2020 or something...